### DISCUSSION TITLE
Support Concurrent Mode introduces serious performance issues with native HTML drag and drop 

### DISCUSSION THREAD

**Discussion by kokipsiho** [vercel/next.js#11393] [Help]
After upgrading to 9.3.1 from 8.x we noticed a performance issue with drag and drop. UI starts to flicker, scroll is not responding etc... Trying to get some data from Chrome performance tools slows browser tab to the point that entire Chrome UI becomes unresponsive. By going back and checking releases we found that issue was introduced with PR #[9026](https://github.com/zeit/next.js/pull/9026/files), Support Concurrent Mode in Loadable.

I would like to understand what was gained by using `use-subscription` lib? According to their [website](https://www.npmjs.com/package/use-subscription), this lib has a limited use case and it's probably better to avoid it for frameworks like Next IMO.

**Comment by timneutkens** [comment]
@devknoll can give more context

**Comment by devknoll** [comment]
@kokipsiho could you share a repro case? I don't think it's expected that this would have any effect on perf in non-Concurrent Mode, so it might be a bug.

**Reply by devknoll** [reply]
Actually, I have a hunch as to what might be happening. I'll push a new PR.

**Reply by kokipsiho** [reply]
Thanks

**Comment by kokipsiho** [comment]
It's part of a larger app we are currently building so I can't share anything sorry. If you want I can create sample app with the same behavior.

**Reply by devknoll** [reply]
A sample app with the same behavior would be perfect, but feel free to wait until after you get a chance to test on the next canary release to see if it was fixed by the PR :-)

**Reply by kokipsiho** [reply]
Sounds good, thanks.

**Reply by devknoll** [reply]
Looks like `v9.3.5-canary.0` was released, so please give it a try and let me know if it fixes the issue you're seeing. Thank you!

**Reply by kokipsiho** [reply]
Unfortunately, this didn't fix the issue. I'm attaching DevTools profile, it might give you hints on what could be the reason for this.

For now, we will use `loadable-components` as that one is working fine. We are getting same expected behaviour we had with `next/dynamic` prior to PR #9026.

[Profile-20200409T110654.json.zip](https://github.com/zeit/next.js/files/4455133/Profile-20200409T110654.json.zip)

**Comment by Mohsan1995** [comment]
Hi,

Do you have any update with this issue ?

**Reply by devknoll** [reply]
Not yet -- please post a reproduction and I'd be happy to look into it more!
